README.md
<p align="center">
  <a href="https://gethyas.com/">
    <img alt="Doks" src="https://hyas.netlify.app/hyas.svg" width="60">
  </a>
</p>

<h1 align="center">
  Hyas
</h1>

<h3 align="center">
  Modern Hugo Starter
</h3>

<p align="center">
  Hyas is a Hugo starter helping you build modern websites that are secure, fast, and SEO-ready — by default.
</p>

<p align="center">
  <a href="https://github.com/h-enk/hyas/blob/master/LICENSE">
    <img src="https://img.shields.io/github/license/h-enk/hyas?style=flat-square" alt="GitHub">
  </a>
  <a href="https://github.com/h-enk/hyas/releases">
    <img src="https://img.shields.io/github/v/release/h-enk/hyas?include_prereleases&style=flat-square"alt="GitHub release (latest SemVer including pre-releases)">
  </a>
  <a href="https://www.npmjs.com/package/@hyas/core">
    <img src="https://img.shields.io/npm/v/@hyas/core?style=flat-square" alt="npm (scoped)">
  </a>
  <a href="https://github.com/h-enk/hyas/actions?query=workflow%3A%22Hyas+CI%22">
    <img src="https://img.shields.io/github/workflow/status/h-enk/hyas/Hyas%20CI/master?style=flat-square" alt="GitHub Workflow Status (branch)">
  </a>
  <a href="https://app.netlify.com/sites/hyas/deploys">
    <img src="https://img.shields.io/netlify/895a161c-86be-48a2-8c57-a8c5d68cd1a4?style=flat-square" alt="Netlify">
  </a>
</p>

![Hyas — Modern Hugo Starter](https://raw.githubusercontent.com/h-enk/hyas/master/images/tn.png)

- [Demo](#demo)
- [Why Corporio?](#why-corporio)
- [Requirements](#requirements)
- [Get Started](#get-started)
  - [Standalone Website](#standalone-website)
    - [1. Create a new site](#1-create-a-new-site)
    - [2. Install dependencies](#2-install-dependencies)
    - [3. Start development server](#3-start-development-server)
  - [As a Theme](#as-a-theme)
    - [1. Create a new site](#1-create-a-new-site-1)
    - [2. Install dependencies](#2-install-dependencies-1)
    - [3. Configure website](#3-configure-website)
- [Other commands](#other-commands)
- [Documentation](#documentation)
- [Communities](#communities)

## Demo

- [corporio.onrender.com](https://corporio.onrender.com/)

## Why Corporio?

Six reasons why you should use Corporio:

1. **Security aware**. Get A+ scores on [Mozilla Observatory](https://observatory.mozilla.org/analyze/hyas.netlify.app) out of the box. Easily change the default Security Headers to suit your needs.
2. **Fast by default**. Get 100 scores on [Google Lighthouse](https://googlechrome.github.io/lighthouse/viewer/?gist=bc1047434376aa5b3e50cf637f8a5b40) by default. Corporio removes unused css, prefetches links, and lazy loads images.
3. **SEO-ready**. Use sensible defaults for structured data, open graph, and Twitter cards. Or easily change the SEO settings to your liking.
4. **Development tools**. Code with confidence. Check styles, scripts, and markdown for errors and fix automatically or manually.
5. **Bootstrap framework**. Build robust, flexible, and intuitive websites with Bootstrap. Or use any other front-end framework if you prefer.
6. **Netlify-ready**. Deploy to Netlify with sensible defaults. Easily use Netlify Functions, Netlify Redirects, and Netlify Headers.

## Requirements

- [Git](https://git-scm.com/) — latest source release
- [Node.js](https://nodejs.org/) — latest LTS version

## Get Started

Start a new Corporio themed site in two ways either as a theme or as a standalone website
Please note that it is recommended to use PNPM however you may also use NPM or Yarn, but they are not guaranteed to work.

### Standalone Website

#### 1. Create a new site

```bash
git clone https://github.com/AminZibayi/Corporio.git my-corporio-site && cd my-corporio-site
```

#### 2. Install dependencies

```bash
npm install
# OR
pnpm install
```

#### 3. Start development server

```bash
npm run start
# OR
pnpm start
```

### As a Theme

#### 1. Create a new site

```bash
git submodule add https://github.com/AminZibayi/Corporio.git themes/corporio
```

#### 2. Install dependencies

```bash
cd /themes/corporio && pnpm install
```

#### 3. Configure website

Add `theme: "corporio"` to your website config file.

## Other commands

_TODO_

## Documentation

_TODO_

## Communities

_TODO_


CHANGELOG.md
### Changelog

All notable changes to this project will be documented in this file. Dates are displayed in UTC.

Generated by [`auto-changelog`](https://github.com/CookPete/auto-changelog).

#### [v1.5.8](https://github.com/h-enk/hyas/compare/v1.5.7...v1.5.8)

- Better image support [`#316`](https://github.com/h-enk/hyas/pull/316)
- esbuild idea [`#308`](https://github.com/h-enk/hyas/pull/308)
- deps: bump versions to latest [`1f7b9ff`](https://github.com/h-enk/hyas/commit/1f7b9fff44a4312d5ad94344e91435e1ba0dfcc7)
- deps: bump versions to latest [`833a497`](https://github.com/h-enk/hyas/commit/833a4978ce9f62253b7c98196bcd4495ac18c809)
- feat: add @hyas/images [`d13eca3`](https://github.com/h-enk/hyas/commit/d13eca33ed89281bfd63532a4f4984d4530a0e8e)

#### [v1.5.7](https://github.com/h-enk/hyas/compare/v1.5.6...v1.5.7)

> 17 May 2022

- deps: bump versions to latest [`e3f95b1`](https://github.com/h-enk/hyas/commit/e3f95b1725f42891b408a227d13cbc5a0c445801)
- meta: update changelog [`f43833f`](https://github.com/h-enk/hyas/commit/f43833f825b22323cc6393f26857f80a9bfe8db2)
- fix: update changelog [`1223796`](https://github.com/h-enk/hyas/commit/1223796b2e1674b29ffecebe773eac15b7ad9e84)

#### [v1.5.6](https://github.com/h-enk/hyas/compare/v1.5.5...v1.5.6)

> 24 March 2022

- deps: bump versions to latest [`1e551b9`](https://github.com/h-enk/hyas/commit/1e551b954e26131f7e8c8e3f697f05dc1c0fd7c5)
- feat: add image partial [`0a7cbc9`](https://github.com/h-enk/hyas/commit/0a7cbc9e95dac76a710f09c539d14257adbcd333)
- meta: add sponsors and backers sections [`f83b583`](https://github.com/h-enk/hyas/commit/f83b583a232287d02c4b54a516298017076d534a)

#### [v1.5.5](https://github.com/h-enk/hyas/compare/v1.5.4...v1.5.5)

> 4 February 2022

- deps: bump versions to latest [`3ea1ae7`](https://github.com/h-enk/hyas/commit/3ea1ae79b026a1cbb77c78cd0fa3e99dfd684b1f)
- deps: replace markdownlint-cli with markdownlint-cli2 [`7131dce`](https://github.com/h-enk/hyas/commit/7131dce421eb028a27c0bc813104123c1e338009)
- meta: update changelog [`ed477ef`](https://github.com/h-enk/hyas/commit/ed477efd5354ea463e34ce34679b6e378cf85f40)

#### [v1.5.4](https://github.com/h-enk/hyas/compare/v1.5.3...v1.5.4)

> 7 January 2022

- feat: add dismissible global alert [`#232`](https://github.com/h-enk/hyas/pull/232)
- deps: bump versions to latest [`dc0da29`](https://github.com/h-enk/hyas/commit/dc0da29b6fa7431efd71630ad7bffa26485e0b22)
- meta: update changelog [`13f9bf7`](https://github.com/h-enk/hyas/commit/13f9bf735789e3375a85a6312ede73a9d985dcf9)
- feat: update dismissible global alert [`13cb865`](https://github.com/h-enk/hyas/commit/13cb86595eef33ccdb72b2d6c26ba23feee5cd8d)

#### [v1.5.3](https://github.com/h-enk/hyas/compare/v1.5.2...v1.5.3)

> 5 November 2021

- fix: remove stray character [`#211`](https://github.com/h-enk/hyas/pull/211)
- feat: replaced shortcode email obfuscation w/ render-link email obfus… [`#195`](https://github.com/h-enk/hyas/pull/195)
- deps: bump versions to latest [`2c3c3a9`](https://github.com/h-enk/hyas/commit/2c3c3a9c5cee4b78875c07e4d5bae5a1eb71d4fe)
- feat: replaced shortcode email obfuscation w/ render-link email obfuscation [`b1b2c0c`](https://github.com/h-enk/hyas/commit/b1b2c0c225c9376c6b849f537ec7605dd76ecbeb)
- meta: update changelog [`52544cd`](https://github.com/h-enk/hyas/commit/52544cdf9d48d0bd6951a219747dbf66911e3df3)

#### [v1.5.2](https://github.com/h-enk/hyas/compare/v1.5.1...v1.5.2)

> 21 September 2021

- feat: replaced image shortcodes w/ image render hook [`#188`](https://github.com/h-enk/hyas/pull/188)
- config: update + remove unused params [`#187`](https://github.com/h-enk/hyas/pull/187)
- fix: change body classes [`#165`](https://github.com/h-enk/hyas/pull/165)
- Remove "shrink-to-fit=no" [`#184`](https://github.com/h-enk/hyas/pull/184)
- deps: bump @babel/cli from 7.15.4 to 7.15.7 [`6554ad4`](https://github.com/h-enk/hyas/commit/6554ad457102cb08887e03172122209d97072b85)
- deps: bump versions to latest [`13fca45`](https://github.com/h-enk/hyas/commit/13fca456ffdbb537c082b391ddac0d265361b90a)
- fix: move body-class to partial [`785380a`](https://github.com/h-enk/hyas/commit/785380a73fd81d109a76a9f4de26a35ce064321c)

#### [v1.5.1](https://github.com/h-enk/hyas/compare/v1.5.0...v1.5.1)

> 27 August 2021

- Replaces instant.page w/ quicklink [`#173`](https://github.com/h-enk/hyas/pull/173)
- Libsass [`#166`](https://github.com/h-enk/hyas/pull/166)
- Esbuild [`#155`](https://github.com/h-enk/hyas/pull/155)
- Using Hugo on Netlify [`#156`](https://github.com/h-enk/hyas/pull/156)
- deps: bump versions to latest [`8f1c6f3`](https://github.com/h-enk/hyas/commit/8f1c6f33b2f7a6e94e6e722cb9a4b2b86b60b912)
- deps: bump autoprefixer + eslint to latest versions [`b144a91`](https://github.com/h-enk/hyas/commit/b144a91582874d8b1b7e73b807bba0f3de04b0b3)
- feat: replaced instant.page w/ quicklink [`2841f42`](https://github.com/h-enk/hyas/commit/2841f4203922e7f28a869f4ba943261bb494288e)

#### [v1.5.0](https://github.com/h-enk/hyas/compare/v1.4.4...v1.5.0)

> 29 June 2021

- feat: add hugo-installer + auto-changelog [`5256d79`](https://github.com/h-enk/hyas/commit/5256d79184e826cf70e155bcd29d070a584a1414)
- fix: add inter font [`7849335`](https://github.com/h-enk/hyas/commit/78493353ce6c5d15dc2ebb39f94ae55d0de60f5e)
- fix: update for stylelint [`6c645d3`](https://github.com/h-enk/hyas/commit/6c645d3d057be868ec80ebbc1c24cb29c9886b73)

#### [v1.4.4](https://github.com/h-enk/hyas/compare/v1.4.3...v1.4.4)

> 25 February 2021

- chore(release): 1.4.4 [`790d480`](https://github.com/h-enk/hyas/commit/790d4808f8edb4b92a9beca350cf750454ff39e7)
- config: update package.json [`632eac8`](https://github.com/h-enk/hyas/commit/632eac8cbfd510dadb019bfd66e20b8333cc6ca2)
- Updating the README! [`27b91e2`](https://github.com/h-enk/hyas/commit/27b91e2aeea831107ad53aa369510f9bc786bac6)

#### [v1.4.3](https://github.com/h-enk/hyas/compare/v1.4.2...v1.4.3)

> 24 February 2021


assets/js/script.js
// In the Name of God, the Originator the Creative

// TODO: don't import whole the modules, only import necessary function when possible
/* TODO: remove jquery plugins and use their alternatives
 * - remove their script tags from footer section
 * - remove them from package.json
 * - remove their styles from libraries.scss
 */
import "./vendor/slick-animation";
import "waypoints/lib/jquery.waypoints";
// TODO: this module is deprecated, use newer alternatives
import { TweenMax, Power3 } from "gsap/TweenMax";
// TODO: remove this module and use lazysizes instead
import "imagesloaded";

// Preloader

function PageLoad() {
  $("body").removeClass("hidden");
  TweenMax.to($(".preloader-text"), 1, {
    force3D: true,
    opacity: 1,
    y: 0,
    delay: 0.2,
    ease: Power3.easeOut,
  });

  var width = 100,
    perfData = window.performance.timing,
    EstimatedTime = -(perfData.loadEventEnd - perfData.navigationStart),
    time = parseInt((EstimatedTime / 500) % 50) * 70;

  // Percentage Increment Animation
  var PercentageID = $("#precent"),
    start = 1,
    end = 100,
    durataion = time;
  animateValue(PercentageID, start, end, durataion);

  function animateValue(id, start, end, duration) {
    var range = end - start,
      current = start,
      increment = end > start ? 1 : -1,
      stepTime = Math.abs(Math.floor(duration / range)),
      obj = $(id);

    var timer = setInterval(function () {
      current += increment;
      $(obj).text(current);
      if (current === end) {
        clearInterval(timer);
      }
    }, stepTime);
  }
  // Fading Out Loadbar on Finised
  setTimeout(function () {
    TweenMax.to($(".percentage, .inner"), 0.7, {
      force3D: true,
      opacity: 0,
      yPercent: -101,
      ease: Power3.easeInOut,
    });
    TweenMax.to($(".preloader-wrap"), 0.7, {
      force3D: true,
      yPercent: -150,
      delay: 0.2,
      ease: Power3.easeInOut,
    });
  }, time);
}
$(document).ready(function () {
  // preloder
  PageLoad();

  // change-navigation-color
  $(window).scroll(function () {
    if ($(document).scrollTop() > 200) {
      $(".navbar").addClass("nav__color__change");
    } else {
      $(".navbar").removeClass("nav__color__change");
    }
  });

  // Smooth scrolling
  var scrollLink = $(".scroll");
  scrollLink.click(function (e) {
    let elem = $(this.hash);
    if (elem.length) {
      e.preventDefault();
      $("body,html").animate(
        {
          scrollTop: elem.offset().top,
        },
        1000
      );
    }
  });

  $(".navbar-nav>li>a").on("click", function () {
    $(".navbar-collapse").collapse("hide");
  });

  // service slider
  $(".service__slider").slick({
    infinite: false,
    slidesToShow: 3,
    slidesToScroll: 1,
    dots: false,
    arrows: false,
    responsive: [
      {
        breakpoint: 992,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 1,
          dots: true,
        },
      },
      {
        breakpoint: 768,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
          dots: true,
        },
      },
    ],
  });

  // skill count
  $(".skill__progress").waypoint(
    function () {
      $(".progress-value span").each(function () {
        $(this)
          .prop("Counter", 0)
          .animate(
            {
              Counter: $(this).text(),
            },
            {
              duration: 3000,
              easing: "swing",
              step: function (now) {
                $(this).text(Math.ceil(now));
              },
            }
          );
      });
      $(".skill__progress_item").addClass("js-animation");
      this.destroy();
    },
    { offset: "80%" }
  );

  // Testimonial slider
  $(".testimonial__slider").slick({
    infinite: true,
    slidesToShow: 2,
    slidesToScroll: 1,
    dots: true,
    arrows: false,
    autoplay: true,
    autoplaySpeed: 2000,
    responsive: [
      {
        breakpoint: 992,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
          infinite: true,
          dots: true,
        },
      },
    ],
  });

  // Modal Popup
  $(".popup-button").magnificPopup({
    disableOn: 700,
    type: "iframe",
    mainClass: "mfp-fade",
    removalDelay: 160,
    preloader: false,

    fixedContentPos: false,
  });

  var productGrid = $(".product-item-grid").masonry({
    itemSelector: ".product-item",
  });

  productGrid.imagesLoaded().progress(function () {
    productGrid.masonry("layout");
  });

  // blob animation
  var tl = new TimelineMax({
    yoyo: true,
    repeat: -1,
  });
  tl.to(".blob", 3, {
    attr: {
      d: "M470.3 133c45.8 42.5 75.3 104.8 60.3 152-15 47.3-74.4 79.6-120.2 110.7-45.8 31.2-78.1 61.3-116.5 67.4-38.4 6.1-83-11.7-110.2-42.8-27.1-31.2-36.9-75.8-44.7-128.1-7.8-52.3-13.5-112.4 13.6-154.9 27.2-42.5 87.3-67.4 148.5-68.5 61.1-1 123.4 21.7 169.2 64.2z",
    },
  })
    .to(".blob", 3, {
      attr: {
        d: "M452.9 141.3c41.2 47 67.6 102.8 56.3 147.4-11.3 44.5-60.4 77.8-101.6 120.6-41.1 42.8-74.4           95.3-117.3 104.9-42.9 9.7-95.4-23.4-122.1-66.2-26.7-42.9-27.4-95.4-32.6-153.2-5.2-57.7-14.8-120.7 11.9-167.7 26.6-47 89.6-78 149-74.5 59.4 3.5 115.2 41.7 156.4 88.7z",
      },
    })
    .to(".blob", 3, {
      attr: {
        d: "M423.5 172.8c30.2 33.9 43.8 80.5 42.9 126.3-.9 45.7-16.5 90.5-46.7 113.1-30.1 22.7-74.9 23.3-124.8 28.3-49.8 5.1-104.7 14.7-146.6-8-41.8-22.7-70.6-77.6-57.8-119.8 12.7-42.2 66.9-71.6 108.7-105.5 41.9-33.8 71.3-72 109.4-80.6 38.1-8.6 84.7 12.4 114.9 46.2z",
      },
    })
    .to(".blob", 3, {
      attr: {
        d: "M455.4 151.1c43.1 36.7 73.4 92.8 60.8 136.3-12.7 43.5-68.1 74.4-111.3 119.4-43.1 45-74 104.1-109.8 109-35.9 5-76.7-44.2-111.8-89.2-35.2-45-64.7-85.8-70.8-132.6-6-46.8 11.6-99.6 46.7-136.3 35.2-36.6 88-57.2 142.4-58.8 54.5-1.7 110.6 15.6 153.8 52.2z",
      },
    });
});

// G-Map
/**
 * Created by Kausar on 06/10/2016.
 */
window.marker = null;

function initialize() {
  var map;
  var lat = $("#map").data("lat");
  var long = $("#map").data("long");
  console.log(lat, long);
  var mapCenter = new google.maps.LatLng(lat, long);
  var style = [
    {
      featureType: "water",
      elementType: "geometry",
      stylers: [
        {
          color: "#e9e9e9",
        },
        {
          lightness: 17,
        },
      ],
    },
    {
      featureType: "landscape",
      elementType: "geometry",
      stylers: [
        {
          color: "#f5f5f5",
        },
        {
          lightness: 20,
        },
      ],
    },
    {
      featureType: "road.highway",
      elementType: "geometry.fill",
      stylers: [
        {
          color: "#ffffff",
        },
        {
          lightness: 17,
        },
      ],
    },
    {
      featureType: "road.highway",
      elementType: "geometry.stroke",
      stylers: [
        {
          color: "#ffffff",
        },
        {
          lightness: 29,
        },
        {
          weight: 0.2,
        },
      ],
    },
    {
      featureType: "road.arterial",
      elementType: "geometry",
      stylers: [
        {
          color: "#ffffff",
        },
        {
          lightness: 18,
        },
      ],
    },
    {
      featureType: "road.local",
      elementType: "geometry",
      stylers: [
        {
          color: "#ffffff",
        },
        {
          lightness: 16,
        },
      ],
    },
    {
      featureType: "poi",
      elementType: "geometry",
      stylers: [
        {
          color: "#f5f5f5",
        },
        {
          lightness: 21,
        },
      ],
    },
    {
      featureType: "poi.park",
      elementType: "geometry",
      stylers: [
        {
          color: "#dedede",
        },
        {
          lightness: 21,
        },
      ],
    },
    {
      elementType: "labels.text.stroke",
      stylers: [
        {
          visibility: "on",
        },
        {
          color: "#ffffff",
        },
        {
          lightness: 16,
        },
      ],
    },
    {
      elementType: "labels.text.fill",
      stylers: [
        {
          saturation: 36,
        },
        {
          color: "#333333",
        },
        {
          lightness: 40,
        },
      ],
    },
    {
      elementType: "labels.icon",
      stylers: [
        {
          visibility: "off",
        },
      ],
    },
    {
      featureType: "transit",
      elementType: "geometry",
      stylers: [
        {
          color: "#f2f2f2",
        },
        {
          lightness: 19,
        },
      ],
    },
    {
      featureType: "administrative",
      elementType: "geometry.fill",
      stylers: [
        {
          color: "#fefefe",
        },
        {
          lightness: 20,
        },
      ],
    },
    {
      featureType: "administrative",
      elementType: "geometry.stroke",
      stylers: [
        {
          color: "#fefefe",
        },
        {
          lightness: 17,
        },
        {
          weight: 1.2,
        },
      ],
    },
  ];
  var mapOptions = {
    // SET THE CENTER
    center: mapCenter,
    // SET THE MAP STYLE & ZOOM LEVEL
    mapTypeId: google.maps.MapTypeId.ROADMAP,
    // REMOVE ALL THE CONTROLS EXCEPT ZOOM
    zoom: 13,
    panControl: false,
    scrollwheel: false,
    zoomControl: true,
    mapTypeControl: false,
    scaleControl: false,
    streetViewControl: false,
    overviewMapControl: false,
    zoomControlOptions: {
      style: google.maps.ZoomControlStyle.LARGE,
    },
  };

  map = new google.maps.Map(document.getElementById("map"), mapOptions);
  // SET THE MAP TYPE
  var mapType = new google.maps.StyledMapType(style, {
    name: "Grayscale",
  });
  map.mapTypes.set("grey", mapType);
  map.setMapTypeId("grey");
  //CREATE A CUSTOM PIN ICON
  var marker_image = $("#map").data("pin");
  var pinIcon = new google.maps.MarkerImage(marker_image, null, null, null, new google.maps.Size(25, 34));
  marker = new google.maps.Marker({
    position: mapCenter,
    map: map,
    icon: pinIcon,
    title: "bizcred",
  });
}

if ($("#map").length > 0) {
  google.maps.event.addDomListener(window, "load", initialize);
}
